src/OVAL/probes/independent/system_info_probe.c

```diff
@@ -565,11 +731,39 @@ int system_info_probe_main(probe_ctx *ctx, void *arg)
 	os_name = architecture = hname = NULL;
 
 #ifdef _WIN32
-	/* TODO: Get system information */
-	os_name = oscap_strdup(unknown);
-	os_version = oscap_strdup(unknown);
-	architecture = oscap_strdup(unknown);
-	hname = oscap_strdup(unknown);
+	WCHAR computer_name_wstr[MAX_COMPUTERNAME_LENGTH + 1];
+	DWORD computer_name_len = MAX_COMPUTERNAME_LENGTH + 1;
+	GetComputerNameW(computer_name_wstr, &computer_name_len);
+
+	SYSTEM_INFO si;
+	GetNativeSystemInfo(&si);
+	const char *arch;
+	switch (si.wProcessorArchitecture) {
+	case PROCESSOR_ARCHITECTURE_AMD64:
```

Comment:
Maybe it would be better to put this out from main function and create something like `static const char *get_windows_architecture(void)`.

Reply:
done